### DISCUSSION TITLE
When installing halo, the string concatenation in compose is incorrect.

### DISCUSSION THREAD

**Discussion by duminghan23** [1Panel-dev/1Panel#7330] [Q&A]
### Contact Information

1844907684@qq.com

### 1Panel Version

v1.10.21-lts

### Problem Description

在compose中的 --spring.r2dbc.url 拼接不正确，没有拼接panel_db_ip、panel_db_port、panel_db_name 这三个变量

### Steps to Reproduce

直接在应用商店安装halo，选择MySQL，使用默认生成的数据库、账号、密码，等待halo启动完成，查看日志，显示连接mysql失败。
经过检查，在compose中的  --spring.r2dbc.url 拼接不正确，没有拼接panel_db_ip、panel_db_port、panel_db_name 这三个变量。
解决方法：手动编写compose文件，把使用$符号拼接的参数手动拼接上，如下，点击确定后就可以正常安装了。
--spring.r2dbc.url=r2dbc:pool:mysql://192.168.2.201:3306/halo_k2csjn

### The expected correct result

正确拼接参数

**Comment by wanghe-fit2cloud** [comment]
Compose 的变量定义在 .env 文件中，你可以进入应用的安装目录，修改相应的变量值后再试一下。

**Comment by wanghe-fit2cloud** [comment]
将 issue 转为 discussion

**Comment by duminghan23** [comment]
我的.env文件内容如下，
``` 
CONTAINER_NAME="myhalo"
CPUS=0
HALO_EXTERNAL_URL="http://localhost:8090"
HOST_IP=""
MEMORY_LIMIT=0
PANEL_APP_PORT_HTTP=8090
PANEL_DB_HOST="mysql"
PANEL_DB_HOST_NAME="mysql"
PANEL_DB_NAME="halo_k2csjn"
PANEL_DB_PORT=3306
PANEL_DB_TYPE="mysql"
PANEL_DB_USER="halo_jWAx54"
PANEL_DB_USER_PASSWORD="halo_NnAjZz"
```
这个里面的参数是不正确，PANEL_DB_HOST应该读取实际的IP地址，如图
![image](https://github.com/user-attachments/assets/913edf54-da35-4ce4-82e7-7ae9f4f91145)
这样，compose文件中的内容：
 --spring.r2dbc.url=r2dbc:pool:${PANEL_DB_TYPE}://${PANEL_DB_HOST}:${PANEL_DB_PORT}/${PANEL_DB_NAME}
 才能够拼接正确

**Reply by wanghe-fit2cloud** [reply]
是因为应用商店中安装的应用使用到了相同的容器网络 `1panel-network`，此时使用容器服务名称也是支持的。
如果存在访问异常的问题，可以尝试编辑 .env 文件来修改 MySQL 数据库 IP。
